src/components/tl/ucp/tl_ucp_team.c

```diff
@@ -28,9 +59,32 @@ UCC_CLASS_INIT_FUNC(ucc_tl_ucp_team_t, ucc_base_context_t *tl_context,
         return UCC_ERR_NOT_SUPPORTED;
     }
 
-    self->preconnect_task    = NULL;
-    self->seq_num            = 0;
-    self->status             = UCC_INPROGRESS;
+    self->preconnect_task = NULL;
+    self->seq_num         = 0;
+    self->status          = UCC_INPROGRESS;
+    self->tuning_str      = "";
+
+    memcpy(&self->cfg, &UCC_TL_UCP_TEAM_LIB(self)->cfg,
+           sizeof(ucc_tl_ucp_team_config_t));
+
+    if (ctx->topo_required) {
+        status = ucc_tl_ucp_get_topo(self);
+        if (UCC_OK != status) {
+            return status;
+        }
+    }
+
+    if (ucc_global_config.file_cfg && !IS_SERVICE_TEAM(self) &&
+        ctx->topo_required && tl_context->lib->use_tuning) {
+        status = ucc_add_team_sections(&self->cfg, ucc_tl_ucp_lib_config_table,
```

Comment:
should we do the same for context config table?

Reply:
@vspetrov , you originally said that this would cause issue and that the correct solution would be to combine context and lib configs - Can you please elaborate on the issue?
Also, in my opinion this can be dealt with in separate PR.

Reply:
resolving for now, we can discuss further in separate PR.